torch_geometric/nn/conv/gat_conv.py

```diff
@@ -176,6 +181,21 @@ def __init__(
             self.lin_edge = None
             self.register_parameter('att_edge', None)
 
+        if residual and concat:
+            if in_channels != out_channels * heads:
+                self.lin_res = Linear(in_channels, out_channels * heads,
```

Comment:
QQ does GAT paper uses this linear transformation if the `in_channels` is different from `out_channels`?

Reply:
No need for linear transformation if the dimension matches.